test/frontend/test_editor.py

```diff
@@ -2,45 +2,304 @@
+
+        self.click_edit_button('affiliations')
+        WebDriverWait(driver, 10).until(
+            EC.element_to_be_clickable((By.XPATH, '//*[@id="affiliations-editor"]'
+                                                  '//button[contains(@class, "button-remove-affiliation-from-author")]'))
+        )
+        driver.find_element_by_xpath('//*[@id="affiliations-editor"]'
+                                     '//button[contains(@class, "button-remove-affiliation-from-author")]').click()
+        driver.find_element_by_xpath('//*[@id="affiliations-editor"]//button[@id="modal-button-ok"]').click()
+
+        assert "Computational" in driver.find_element_by_xpath('//*[@class="affiliations"]//li[1]/span').text
+
+    def test_text(self):
+        driver = self.driver
+        self.click_edit_button('abstract-text')
+
+        WebDriverWait(driver, 30).until(
+            EC.visibility_of_element_located((By.XPATH, '//*[@id="abstract-text-editor"]'))
+        )
+
+        text = driver.find_element_by_xpath('//*[@id="abstract-text-editor"]//textarea[@id="text"]')
+        text.send_keys('Test abstract test.')
+        driver.find_element(By.XPATH, '//*[@id="abstract-text-editor"]//button[@id="modal-button-ok"]').click()
+
+        assert "Test abstract test." in driver.find_element_by_xpath('//*[@class="abstract-text"]/p').text
+
+    def test_acknowledgements(self):
+        driver = self.driver
+        self.click_edit_button('acknowledgements')
+
+        WebDriverWait(driver, 30).until(
+            EC.visibility_of_element_located((By.XPATH, '//*[@id="acknowledgements-editor"]'))
+        )
+
+        acknowledgements = driver.find_element_by_xpath('//*[@id="acknowledgements-editor"]'
+                                                        '//textarea[@id="acknowledgements"]')
+        acknowledgements.send_keys('Thanks.')
+        driver.find_element(By.XPATH, '//*[@id="acknowledgements-editor"]//button[@id="modal-button-ok"]').click()
+
+        assert "Thanks." in driver.find_element_by_xpath('//*[@class="acknowledgements"]/p').text
+
+    def test_add_references(self):
+        driver = self.driver
+        self.click_edit_button('references')
+
+        WebDriverWait(driver, 10).until(
+            EC.element_to_be_clickable((By.XPATH, '//*[@id="references-editor"]//button[contains(@class, "btn-add")]'))
+        )
+
+        driver.find_element_by_xpath('//*[@id="references-editor"]//button[contains(@class, "btn-add")]').click()
+        driver.find_element_by_xpath('//*[@id="references-editor"]'
+                                     '//input[contains(@class, "citation")]').send_keys("John, title.")
+        driver.find_element_by_xpath('//*[@id="references-editor"]'
+                                     '//input[contains(@class, "link")]').send_keys("www.link.com")
+        driver.find_element_by_xpath('//*[@id="references-editor"]'
+                                     '//input[contains(@class, "doi")]').send_keys("12345")
```

Comment:
DOI has to be alphanumeric, otherwise the reference will not be added. Please update accordingly.